src/rsz/include/rsz/Resizer.hh

```diff
@@ -182,6 +182,34 @@ struct BufferData
 
 class OdbCallBack;
 
+// RAII mechanism to facilitate using operation specific settings, e.g.,
+// use only the clock buffers from cts when doing repairClkNets, don't
+// use buffers marked as "clock" in hold repairing, etc.
+// Can be instantiated in two ways:
+// - Before resizePreamble(): set the "filters" used by the preamble.
+// - After resizePreamble(): enforce the variables we want.
+class OperationSettings
```

Comment:
Can this be written in a more generic fashion like
```
template<typename T>
class SetAndRestore
{
   SetAndRestore(const T& new_value, const T& storage); // store current value in storage and assign it to new_value
  ~SetAndRestore(); // assign storage back to its previous value
};
```
Then apply it to the specific cases of exclude_clock_buffers and buffer_cells.